api/cluster/resource/templater.go

```diff
@@ -152,10 +174,34 @@ func (t *InferenceServiceTemplater) PatchInferenceServiceSpec(orig *kservev1beta
 	}
 	orig.ObjectMeta.Annotations = utils.MergeMaps(utils.ExcludeKeys(orig.ObjectMeta.Annotations, configAnnotationKeys), annotations)
 	orig.Spec.Predictor = createPredictorSpec(modelService, config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create predictor spec: %w", err)
+	}
+	orig.Spec.Predictor.TopologySpreadConstraints, err = updateExistingInferenceServiceTopologySpreadConstraints(
+		orig,
+		modelService,
+		config,
+		kservev1beta1.PredictorComponent,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create predictor topology spread constraints: %w", err)
+	}
 
 	orig.Spec.Transformer = nil
 	if modelService.Transformer != nil && modelService.Transformer.Enabled {
-		orig.Spec.Transformer = t.createTransformerSpec(modelService, modelService.Transformer, config)
+		orig.Spec.Transformer = t.createTransformerSpec(modelService, modelService.Transformer)
+		if err != nil {
```

Comment:
We can remove this since t.createTransformerSpec does not return error

Reply:
Oh good catch; I think the error handling was a remnant of a previous iteration of this PR 😅 I forgot to clean those up after I changed the implementation. Thanks for picking this up!